extensions/widgets/treeview/treeview.lcb

```diff
@@ -336,19 +339,40 @@ private handler paintDataItem(in pDataItem as Array, in pTop as Real, in pRow as
 		
 	// Draw the key
 	set the paint of this canvas to getPaint("text","disabled" & tStyle)
-	fill text "[" & pDataItem["key"] & "]" at left of rectangle [tLeft, pTop, mViewWidth - 3 * mMargin - 2 * mIconWidth, pTop + mRowHeight] on this canvas
+
+    // AL-2015-07-26: [[ Bug 15752 ]] Only draw specified amount of key string
+    variable tKeyLength as Number
+    variable tKeyDisplay as String
+    put the number of chars in pDataItem["key"] into tKeyLength
+    if tKeyLength <= kMaxKeyDisplayChars then
+        put pDataItem["key"] into tKeyDisplay
+    else
+        put (char 1 to kMaxKeyDisplayChars of pDataItem["key"]) & "..." into tKeyDisplay
+    end if
```

Comment:
In theory, but as the max is fixed at the moment it doesn't make much difference.

This is a bit of a stopgap until the widget can do some measuring to see how much it can render (possibly with adjustable tab stop)